cfme/tests/infrastructure/test_host.py

```diff
@@ -426,3 +426,28 @@ def test_infrastructure_hosts_navigation_after_download(
     hosts_view.navigation.select("Compute")
     provider_view = provider.create_view(InfraProviderDetailsView)
     assert provider_view.is_displayed
+
+
+@test_requirements.infra_hosts
+@pytest.mark.meta(blockers=[BZ(1746214, forced_streams=["5.10"])], automates=[1746214])
+def test_compare_hosts_from_provider_allhosts(appliance, setup_provider_min_hosts, provider):
+    """
+    Polarion:
+        assignee: prichard
+        casecomponent: Infra
+        caseimportance: high
+        initialEstimate: 1/6h
+    Bugzilla:
+        1746214
+
+    """
+    hosts_view = navigate_to(provider.collections.hosts, "All")
+    num_hosts = len(hosts_view.entities.get_all())
+    if num_hosts < 2:
+        pytest.skip('not enough hosts in appliance UI to run test')
+    for h in hosts_view.entities.get_all():
+        h.check()
+    hosts_view.toolbar.configuration.item_select('Compare Selected items',
+                                                 handle_alert=True)
+    compare_hosts_view = provider.create_view(HostsCompareView)
```

Comment:
For the purposes of the test case, this is okay. But in the future it may be nice to register this view against the collection of hosts (i.e. `provider.collections.hosts`), that way you'd be able to do

```python,
view = navigate_to(provider.collections.hosts, "Compare")
```

Things will get tricky when you want to compare specific sets of hosts, so a method for filtering will need to be implemented (which could be achieved via a collection's filter, checkout e.g. https://github.com/ManageIQ/integration_tests/blob/bf0e545e1f76a2e8f43443f9d0afa123f95e3837/cfme/infrastructure/virtual_machines.py#L1626-L1641). 

I wouldn't consider this necessary for this PR, but if you end up writing a lot of test cases for this page, it will reduce repeated code.

Reply:
Yes. Your suggestions will likely be part of upcoming PRs.